### ISSUE TITLE
[Feature Request] Add support for ResizeObserver to old browsers

### ISSUE THREAD

**Issue by Richard602** [Suwayomi/Suwayomi-WebUI#799]
## What feature should be added to Suwayomi?
Please consider providing legacy support for the ResizeObserver in old browsers. Whether by polyfill or through some other method.

##  Why/Project's Benefit/Existing Problem
This will continue support for old browsers that do not yet have ResizeObserver.

On a iPad stuck at version 12, the browsers (Safari, Chrome, Firefox) cannot be updated to a version that supports this feature, and instead the error `Can't find variable: ResizeObserver` will pop up.

I've tried rebuilding the v1.1.0 release with polyfill from @juggle/resize-observer , and this seems to have successfully added it to the version of Safari i'm running, so I believe only the ResizeObserver feature is missing to run the WebUI on old browsers.

**Comment by schroda** [comment]
I don't have a way to test this myself, but prs are always welcome

**Comment by Richard602** [comment]
Hi. I was also hoping to use vite plugin legacy. While assuming I can just add a line into `vite.config.ts` to get it, I'm not sure what to put inside `modernPolyfills: []`, I don't how to look for the string value that I can put in there to get ResizeObserver. From the little searching I did, I can't seem to find or see vite plugin legacy being able to provide ResizeObserver.

Also, I instead end up finding [this dev.to post/article](https://dev.to/solleedata/supporting-older-browsers-using-vite-2ii#-raw-resizeobserver-endraw-bug) which said `ResizeObserver` needs `additionalLegacyPolyfills: []` instead.

Sorry for bringing up this hassle. I've only looked into vite the day I made the feature request, but If you can point me towards where I need to properly look to get vite plugin legacy to provide ResizeObserver, I'll be happy to try it out and see if it can work.

**Comment by schroda** [comment]
from my understanding `additionalLegacyPolyfills` takes a npm package (e.g. `@juggle/resize-observer`) which needs to auto polyfill, which, from what I've seen, `@juggle/resize-observer` does not do, so I guess vite won't work and you have to manually polyfill it

**Comment by Richard602** [comment]
A pity. Oh well. My feature request still stands, in hopes of a method other than using `@juggle/resize-observer` appears.

**Comment by schroda** [comment]
I've just added `@juggle/resize-observer` manually without vite (f7e5388e2a60a55be1230a4b05fd5c1698721c9a)

**Comment by Richard602** [comment]
Pardon me for the late response. tried building from source to see if it works, but it seems I can't get past the login on my ipad. I guess I'll try again once the new UI is released in a major release.

**Comment by schroda** [comment]
are you using the stable server?
for the latest webUI version you have to use the servers preview version, so this could be the problem

**Comment by Richard602** [comment]
I was using the stable version indeed. I'll try again at some point with the server preview version. For now, i'll edit my previous comment to describe what happens when using the stable server.

**Comment by Richard602** [comment]
Hello,
Took me some time to get the little bit of free time to try out the preview versions.
This time, I am `Running Suwayomi-Server v1.1.1 revision r1599`, both docker and... dev run?

With the docker server, I switched the WebUI to the preview channel in the GUI, which made it use ` (version= r1923, flavor= WebUI)` according to the terminal.
This still has the same issue; works great on pc browser, white screen on ipad

Also tried cloning the server repo and running it with `./gradlew server:run --stacktrace`, then building the webui with `yarn build` after cloning the repo. Same result; works great on pc browser, white screen on ipad

Happy to try again if I missed something, though again, might take time for me to try it out.

P.S. New WebUI is looking pretty good~ Feels a touch more modern compared to the current stable version
P.P.S I probably don't need to say it here, but just wanna say that I had switch `com.github.inorichi.injekt:injekt-core:65b0440` to `uy.kohesive.injekt:injekt-core:1.16.1` to successfully run the server after cloning the server repo.
P.P.P.S I should've mentioned this earlier, but this is what I added to App.tsx when I first mentioned that I rebuilt the v1.1.0
```
import { ResizeObserver as Polyfill } from '@juggle/resize-observer';

window.ResizeObserver = window.ResizeObserver || Polyfill;
const ResizeObserver = window.ResizeObserver || Polyfill;

// Uses native or polyfill, depending on browser support.
const ro = new ResizeObserver((entries, observer) => {
  console.log('Something has resized!');
});

window.ResizeObserver = window.ResizeObserver || Polyfill;
```
On that note, I'll try to see if that works with the preview version as well, then go back to edit this comment with the result.

**Comment by schroda** [comment]
from what I've found, the safari version, that comes with iOS 12, only supports ES5.
I've found my old iPhone 5s which also runs iOS 12 and there it's not even able to execute the index.tsx file

which is weird, because you were apparently able to run the app with just adding a polyfill for the ResizeObserver while using an iPad which also runs iOS 12?

**Comment by Richard602** [comment]
Yep, not quite sure how to check the exact model, but in Settings it says it's an iPad Air with software version 12.5.7.

It was a bit weird as well. If I recall correctly, only using the code snippet in [juggle's page](https://github.com/juggle/resize-observer?tab=readme-ov-file#basic-usage) didn't work, and neither did using the snippet from [that dev.to article](https://dev.to/solleedata/supporting-older-browsers-using-vite-2ii#-raw-resizeobserver-endraw-bug). Combining the two however (as you may or may not have seen in my comment's edit history), for some reason worked and is what I'm using at the moment. So yeah, it for some reason just works by adding those lines into App.tsx for me.

A bit unrelated but correct me if i'm wrong, the device shouldn't be running any .tsx files yeah? at least I don't see any .tsx files inside the build or webUI folder after running `yarn build`...

**Comment by schroda** [comment]
tsx gets compiled to javascript, and the version is gets compiled to is not supported by safari of iOS 12, so it shouldn't be possible to be running on these systems

to get it to work you'll have to update it to compile to ES5

**Comment by Richard602** [comment]
Ah so that's what you mean.

And so that's what you mean that somehow the previous webUI version is working despite the recompile/rebuild.

I guess I'll be googling around how to do that in my free time to see if that can get things working. Thank you very much.

**Comment by schroda** [comment]
I do not plan to fix this. But you (or any one else) are welcome to open a pr for it
